libraries/fc/include/fc/static_variant.hpp

```diff
@@ -14,587 +14,112 @@
 #include <typeinfo>
 #include <fc/exception/exception.hpp>
 #include <boost/core/typeinfo.hpp>
+#include <variant>
 
 namespace fc {
 
-// Implementation details, the user should not import this:
-namespace impl {
-
-template<typename R, int N, typename... Ts>
-struct storage_ops;
-
-template<typename X, typename... Ts>
-struct position;
-
-template<int Pos, typename... Ts>
-struct type_at;
+template<typename Result>
+struct visitor {};
 
-template<typename... Ts>
-struct type_info;
+template<typename T, typename... Types>
+constexpr T& get(std::variant<Types...>& v)
```

Comment:
Why are you not just using `std::get` ? Add comment if there is a good reason.

Reply:
@allenhan2 mentioned trying to maintain the same exception semantics.  So instead of a `std::bad_variant_access` exception being raised on an unchecked call to `std::get`, it would be better to raise the same `fc::assert_exception` exception so I created `fc::get` to mimic `std::get` but raise the fc exception.